cmd/endpoint_test.go

```diff
@@ -38,42 +38,46 @@ func TestNewEndpoint(t *testing.T) {
+		{"/foo", Endpoint{URL: &url.URL{Path: "/foo"}, IsLocal: true}, PathEndpointType, nil, false},
+		{`\foo`, Endpoint{URL: &url.URL{Path: `\foo`}, IsLocal: true}, PathEndpointType, nil, false},
+		{"C", Endpoint{URL: &url.URL{Path: `C`}, IsLocal: true}, PathEndpointType, nil, false},
+		{"C:", Endpoint{URL: &url.URL{Path: `C:`}, IsLocal: true}, PathEndpointType, nil, false},
+		{"C:/", Endpoint{URL: &url.URL{Path: "C:"}, IsLocal: true}, PathEndpointType, nil, false},
+		{`C:\`, Endpoint{URL: &url.URL{Path: `C:\`}, IsLocal: true}, PathEndpointType, nil, false},
+		{`C:\foo`, Endpoint{URL: &url.URL{Path: `C:\foo`}, IsLocal: true}, PathEndpointType, nil, false},
+		{"C:/foo", Endpoint{URL: &url.URL{Path: "C:/foo"}, IsLocal: true}, PathEndpointType, nil, false},
+		{`C:\\foo`, Endpoint{URL: &url.URL{Path: `C:\\foo`}, IsLocal: true}, PathEndpointType, nil, false},
+		{"http:path", Endpoint{URL: &url.URL{Path: "http:path"}, IsLocal: true}, PathEndpointType, nil, false},
+		{"http:/path", Endpoint{URL: &url.URL{Path: "http:/path"}, IsLocal: true}, PathEndpointType, nil, false},
+		{"http:///path", Endpoint{URL: &url.URL{Path: "http:/path"}, IsLocal: true}, PathEndpointType, nil, false},
+		{"http://localhost/path", Endpoint{URL: u1, IsLocal: true}, URLEndpointType, nil, false},
+		{"http://localhost/path//", Endpoint{URL: u1, IsLocal: true}, URLEndpointType, nil, false},
+		{"https://example.org/path", Endpoint{URL: u2}, URLEndpointType, nil, false},
+		{"http://127.0.0.1:8080/path", Endpoint{URL: u3, IsLocal: true}, URLEndpointType, nil, false},
+		{"http://192.168.253.200/path", Endpoint{URL: u4}, URLEndpointType, nil, false},
+		{"", Endpoint{}, -1, fmt.Errorf("empty or root endpoint is not supported"), false},
+		{"/", Endpoint{}, -1, fmt.Errorf("empty or root endpoint is not supported"), false},
+		{`\`, Endpoint{}, -1, fmt.Errorf("empty or root endpoint is not supported"), false},
+		{"c://foo", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format"), false},
+		{"ftp://foo", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format"), false},
+		{"http://server/path?location", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format"), false},
+		{"http://:/path", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format: invalid port number"), false},
+		{"http://:8080/path", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format: empty host name"), false},
+		{"http://server:/path", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format: invalid port number"), false},
+		{"https://93.184.216.34:808080/path", Endpoint{}, -1, fmt.Errorf("invalid URL endpoint format: port number must be between 1 to 65535"), false},
+		{"http://server:8080//", Endpoint{}, -1, fmt.Errorf("empty or root path is not supported in URL endpoint"), false},
+		{"http://server:8080/", Endpoint{}, -1, fmt.Errorf("empty or root path is not supported in URL endpoint"), false},
+		{"http://server/path", Endpoint{}, -1, fmt.Errorf("lookup server" + errMsg), true},
```

Comment:
@ws141 
> Tests TestNewEndpointand TestGetHostIP freeze inside a infinite loop in getHostIP4() when ran inside a docker container.

Yes. `getHostIP4()` tries infinitely to resolve host name in Docker or Kubernetes.

As test case `"http://server/path` only skipped for docker/kubernetes environment, you could take out this case from this structure and have another structure with cases only on non-docker environment.

`skipDocker` doesn't look neat.

Reply:
More specifically you could the test case only on non-docker environment